artifactory_cleanup/artifactorycleanup.py

```diff
@@ -56,9 +59,15 @@ def cleanup(self, block_ctx_mgr, test_ctx_mgr) -> Iterator[CleanupSummary]:
                 print(f"Found {len(artifacts_to_remove)} artifacts AFTER filtering")
 
                 # Delete artifacts
-                for artifact in artifacts_to_remove:
+                def _delete_with_context(artifact, policy, test_ctx_mgr, get_name_for_ci, destroy, ignore_not_found):
```

Comment:
only `artifact` argument is required here as I can see, so the signature should be something like 
All other methods are imported, flags we can get from `self` as well.

```python
def delete(self, artifact):
    with test_ctx_mgr(get_name_for_ci(artifact)):
        policy.delete(artifact, destroy=self.destroy)

executor.submit(self._delete, artifact=artifact)
```

Also we could switch to https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.Executor.map

```python
executor.map(self._delete, artifacts_to_remove)
```


---

P.S.
Well, it'll create mess in output anyway, because all thread will print in the same output... But it's the useful case for people who doesn't care about CI, agree.
The task is not so easy if we think about that, especially in case where you use it in CI.